src/components/pathway/PathwayModal.jsx

```diff
@@ -64,12 +65,37 @@ const renderStepNodes = (step, slug) => [].concat(step.courses, step.programs).m
 });
 
 const PathwayModal = ({ learnerPathwayUuid, isOpen, onClose }) => {
+  const intl = useIntl();
+
   const { enterpriseConfig: { slug } } = useContext(AppContext);
   const pathwayUuid = isOpen ? learnerPathwayUuid : null;
   const [pathway, isLoading] = useLearnerPathwayData({ learnerPathwayUuid: pathwayUuid });
 
   if (isOpen === false) { return null; }
 
+  const getPathwayStepsTitle = (step, index) => {
+    const requirementsText = intl.formatMessage({
+      id: 'enterprise.pathway.modal.pathway.requirements.step.title',
+      defaultMessage: 'Requirement {index}',
+      description: 'Title indicating the requirements step for a pathway on the pathway modal.',
+    }, {
+      index: index + 1,
+    });
+
+    const totalCoursesAndPrograms = (step?.courses?.length || 0) + (step?.programs?.length || 0);
+    const chooseAnyOptionText = totalCoursesAndPrograms > 1
+      ? intl.formatMessage({
+        id: 'enterprise.pathway.modal.pathway.requirements.step.choose.any.pathway.text',
+        defaultMessage: ': Choose any {count} of the following',
+        description: 'Text for choosing programs/courses from available options on the pathway modal',
+      }, {
+        count: totalCoursesAndPrograms,
+      })
+      : '';
+
+    return `${requirementsText}${chooseAnyOptionText}`;
```

Comment:
I think we can refactor this code use the logic that I suggested in `src/components/pathway-progress/PathwayStep.jsx` similar to below

```javascript
const requirementsTextWithSingleNode = intl.formatMessage(
  {
    id: 'enterprise.dashboard.pathways.progress.page.pathway.requirements.step.title',
    defaultMessage: 'Requirement {index}',
    description: 'Title indicating the requirements step for a pathway on the pathway progress page.',
  },
  {
    index: index + 1,
  },
);

const requirementsTextWithMultipleNodes = intl.formatMessage(
  {
    id: 'enterprise.dashboard.pathways.progress.page.pathway.requirements.step.title',
    defaultMessage: 'Requirement {index}: Choose any {count} of the following',
    description: 'Title indicating the requirements step for a pathway on the pathway progress page.',
  },
  {
    index: index + 1,
    count: nodes?.length,
  },
);

const title = nodes?.length > 1 ? requirementsTextWithMultipleNodes : requirementsTextWithSingleNode;
```